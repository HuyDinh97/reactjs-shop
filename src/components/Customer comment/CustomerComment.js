import React from 'react';
import { Swiper, SwiperSlide } from 'swiper/react';
import { Pagination } from 'swiper';
import useFetchTestimotional from 'Hooks/useFetchTestimotional';
import quoteImg from './image/icons8-quote-left-48.png';

import 'swiper/swiper-bundle.min.css';
import 'swiper/swiper.min.css';

import classes from './CustomerComment.module.css';

function CustomerComment() {
  const { testimotional, isSuccess } = useFetchTestimotional();

  if (isSuccess === false) {
    return <p data-testid="error-fetch-blog">Error, cannot fetch API</p>;
  }

  if (testimotional?.length <= 0) {
    return <p>Loading...</p>;
  }
  const linkIMG = 'https://vnguyen.xyz/huy/day17/apis/';
  return (
    <div>
      <Swiper
        pagination
        modules={[Pagination]}
        className="mySwiper"
        data-testid="testimotional-element"
      >
        {testimotional &&
          testimotional.map((testi) => {
            return (
              <SwiperSlide
                className={classes.Box}
                key={testi._id.$oid.toString()}
              >
                <div>
                  <div className={classes.Box_inside}>
                    <div className={classes.Box_avatar}>
                      <img src={linkIMG + testi.avatar} alt="img" />
                    </div>
                    <div>
                      <div>
                        <img
                          className={classes.quote_w}
                          src={quoteImg}
                          alt="quoteImg"
                        />
                      </div>
                      <div>
                        <p className={classes.user_comment}>
                          {testi.feedback_text}
                        </p>
                      </div>
                      <div>
                        <p className={classes.user_name}>
                          {' - '}
                          {testi.customer_name}
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </SwiperSlide>
            );
          })}
      </Swiper>
    </div>
  );
}

export default CustomerComment;
