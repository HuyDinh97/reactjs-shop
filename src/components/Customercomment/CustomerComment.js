import React from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { Swiper, SwiperSlide } from 'swiper/react';
import { Pagination } from 'swiper';
import { useGetHome } from 'store/selectors/common';
import quoteImg from './image/icons8-quote-left-48.png';

import 'swiper/swiper-bundle.min.css';
import 'swiper/swiper.min.css';

import classes from './CustomerComment.module.css';

function CustomerComment() {
  const homeData = useGetHome();
  const { testimonial: testimonials } = homeData ?? {};

  if (!testimonials) {
    return <p data-testid="testimotional-error">Loading...</p>;
  }

  const linkIMG = 'https://vnguyen.xyz/huy/day17/apis/';

  return (
    <div>
      <Swiper
        pagination={{
          clickable: true,
          bulletClass: `swiper-pagination-bullet swiper-pagination-testClass`,
        }}
        modules={[Pagination]}
        className={classes.mySwiper}
        data-testid="testimotional-element"
      >
        {testimonials &&
          testimonials.map((testi) => (
            <SwiperSlide className={classes.Box} key={testi._id.$oid}>
              <Row className="mt-4 py-5">
                <Col xl={3} xs={12} className={classes.Box_avatar}>
                  <img src={linkIMG + testi.avatar} alt="img" />
                </Col>
                <Col xl={9} xs={12} className="px-5">
                  <div>
                    <img
                      className={classes.quote_w}
                      src={quoteImg}
                      alt="quoteImg"
                    />
                  </div>
                  <div>
                    <p className={classes.user_comment}>
                      {testi.feedback_text}
                    </p>
                  </div>
                  <div>
                    <p className={classes.user_name}>
                      {' - '}
                      {testi.customer_name}
                    </p>
                  </div>
                </Col>
              </Row>
            </SwiperSlide>
          ))}
      </Swiper>
    </div>
  );
}

export default CustomerComment;
