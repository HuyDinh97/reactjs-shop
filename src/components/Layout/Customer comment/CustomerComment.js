import React from 'react';
import { Swiper, SwiperSlide } from 'swiper/react';
import { Pagination } from 'swiper';
<<<<<<< HEAD
import useFetchBlog from 'Hooks/useFetchBlog';
=======
import useFetchTestimotional from 'Hooks/useFetchTestimotional';
>>>>>>> 6df556a645fdd993207090b36585764d536f464c
import quoteImg from './image/icons8-quote-left-48.png';

import 'swiper/swiper-bundle.min.css';
import 'swiper/swiper.min.css';

import classes from './CustomerComment.module.css';

function CustomerComment() {
<<<<<<< HEAD
  const { blogs, isSuccess } = useFetchBlog(); // Destructuring

  // Safe code: handle if API error in response
  if (isSuccess === false) {
    return <p>Error, cannot fetch API</p>;
  }

  if (blogs.length <= 0) {
    return <p>Loading...</p>;
  }
  return (
    <div>
      <Swiper pagination modules={[Pagination]} className="mySwiper">
        {blogs &&
          blogs.map((blog) => (
            <SwiperSlide className={classes.Box} key={blog._id}>
              <div>
                <div className={classes.Box_inside}>
                  <div>
                    <img src={blog.image} alt="img" />
                  </div>
                  <div>
                    <div>
                      <img
                        className={classes.quote_w}
                        src={quoteImg}
                        alt="quoteImg"
                      />
                    </div>
                    <div>
                      <p className={classes.user_comment}>
                        {blog.content.substring(0, 320)}
                      </p>
                    </div>
                    <div>
                      <p className={classes.user_name}>{blog.author}</p>
                    </div>
                  </div>
                </div>
              </div>
            </SwiperSlide>
          ))}
=======
  const { testimotional, isSuccess } = useFetchTestimotional();

  if (isSuccess === false) {
    return <p data-testid="error-fetch-blog">Error, cannot fetch API</p>;
  }

  if (testimotional?.length <= 0) {
    return <p>Loading...</p>;
  }
  const linkIMG = 'https://vnguyen.xyz/huy/day17/apis/';
  return (
    <div>
      <Swiper
        pagination
        modules={[Pagination]}
        className="mySwiper"
        data-testid="testimotional-element"
      >
        {testimotional &&
          testimotional.map((testi) => {
            return (
              <SwiperSlide
                className={classes.Box}
                key={testi._id.$oid.toString()}
              >
                <div>
                  <div className={classes.Box_inside}>
                    <div className={classes.Box_avatar}>
                      <img src={linkIMG + testi.avatar} alt="img" />
                    </div>
                    <div>
                      <div>
                        <img
                          className={classes.quote_w}
                          src={quoteImg}
                          alt="quoteImg"
                        />
                      </div>
                      <div>
                        <p className={classes.user_comment}>
                          {testi.feedback_text}
                        </p>
                      </div>
                      <div>
                        <p className={classes.user_name}>
                          {' - '}
                          {testi.customer_name}
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </SwiperSlide>
            );
          })}
>>>>>>> 6df556a645fdd993207090b36585764d536f464c
      </Swiper>
    </div>
  );
}

export default CustomerComment;
