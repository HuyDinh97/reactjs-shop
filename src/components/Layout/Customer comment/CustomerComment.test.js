import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import CustomerComment from './CustomerComment';

test('should CustomerComment run correctly', () => {
  const Wrapper = render(<CustomerComment />);

  expect(Wrapper).toMatchSnapshot();
});
