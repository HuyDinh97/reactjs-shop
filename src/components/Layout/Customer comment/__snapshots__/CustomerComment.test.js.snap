// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should CustomerComment run correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="swiper swiper-initialized swiper-horizontal swiper-pointer-events mySwiper"
        >
          <div
            class="swiper-wrapper"
            style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
          >
            <div
              class="swiper-slide"
            >
              <div>
                <div>
                  <div>
                    <img
                      alt="img"
                      src="test-file-stub"
                    />
                  </div>
                  <div>
                    <div>
                      <img
                        alt="quoteImg"
                        src="test-file-stub"
                      />
                    </div>
                    <div>
                      <p>
                        asdasdasdasdasd
                      </p>
                    </div>
                    <div>
                      <p>
                        - William Genske
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="swiper-slide"
            >
              Slide 2
            </div>
            <div
              class="swiper-slide"
            >
              Slide 3
            </div>
          </div>
          <div
            class="swiper-pagination swiper-pagination-bullets swiper-pagination-horizontal"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="swiper swiper-initialized swiper-horizontal swiper-pointer-events mySwiper"
      >
        <div
          class="swiper-wrapper"
          style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
        >
          <div
            class="swiper-slide"
          >
            <div>
              <div>
                <div>
                  <img
                    alt="img"
                    src="test-file-stub"
                  />
                </div>
                <div>
                  <div>
                    <img
                      alt="quoteImg"
                      src="test-file-stub"
                    />
                  </div>
                  <div>
                    <p>
                      asdasdasdasdasd
                    </p>
                  </div>
                  <div>
                    <p>
                      - William Genske
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide"
          >
            Slide 2
          </div>
          <div
            class="swiper-slide"
          >
            Slide 3
          </div>
        </div>
        <div
          class="swiper-pagination swiper-pagination-bullets swiper-pagination-horizontal"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
